dango/lending/src/execute.rs

```diff
@@ -41,13 +50,52 @@ fn update_markets(
         "Only the owner can whitelist denoms"
     );
 
-    for (denom, _updates) in updates {
-        MARKETS.save(ctx.storage, &denom, &Market {})?;
+    for (denom, updates) in updates {
+        if let Some(current_market) = MARKETS.may_load(ctx.storage, &denom)? {
+            let market = updates.build(current_market);
+            MARKETS.save(ctx.storage, &denom, &market)?;
+        } else {
+            let market = Market::try_from(updates)
+                .map_err(|_| anyhow!("Updates must contain all fields to create a new market"))?;
+            MARKETS.save(ctx.storage, &denom, &market)?;
+        }
     }
 
     Ok(Response::new())
 }
 
+pub fn set_collateral_power(
+    ctx: MutableCtx,
+    denom: Denom,
+    power: CollateralPower,
+) -> anyhow::Result<Response> {
+    // Ensure only chain owner can update collateral powers.
+    ensure!(
+        ctx.sender == ctx.querier.query_config()?.owner,
+        "Only the owner can update collateral powers"
+    );
+
+    let mut collateral_powers = COLLATERAL_POWERS.load(ctx.storage)?;
+    collateral_powers.insert(denom, power);
+    COLLATERAL_POWERS.save(ctx.storage, &collateral_powers)?;
```

Comment:
use `COLLATERAL_POWERS.may_update`?

Reply:
Do you prefer may_update and remove setting the Item to an empty BTreeMap in instantiate entrypoint? Or fine to use update and assume it always exists?